### DISCUSSION TITLE
ggwaffle StatWaffleIron extension?

### DISCUSSION THREAD

**Discussion by EvaMaeRey** [ggplot2-extenders/ggplot-extension-club#45] [General]
Browsing some extensions, and wondering if compute could be moved to a Stat (or position_?):

ibrary(ggwaffle)
#> Loading required package: ggplot2
waffle_data <- waffle_iron(mpg, aes_d(group = class))

ggplot(waffle_data, aes(x, y, fill = group)) + 
  geom_waffle()  # uses stat identity and GeomWaffleSquare/Circle

There is a statement about computation so maybe this has been explored:
'Because of the inner mechanisms of ggplot2, some of the necessary data transformations have to be completed outside of a standard plot creation. The function waffle_iron has been added to help with issue.' 

geom_waffle <- function(
  mapping = NULL,
  data = NULL,
  stat = "identity",
  position = "identity",
  tile_shape = c("square", "circle"),
  ...,
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE
){
  tile_shape <- match.arg(tile_shape)
  layer(
    data = data,
    mapping = mapping,
    stat = stat,
    geom = switch(
      tile_shape,
      square = GeomWaffleSquare,
      circle = GeomWaffleCircle
    ),
    position = position,
    show.legend = show.legend,
    inherit.aes = inherit.aes,
    params = list(
      na.rm = na.rm,
      ...
    )
  )
}

**Comment by teunbrand** [comment]
Had a quick glance, and yeah, seems like their [`waffle_iron()`](https://github.com/liamgilbey/ggwaffle/blob/master/R/iron.R) could be moved to a Stat. Might make using the package more comfortable.

**Reply by EvaMaeRey** [reply]
Also worth noting that the {waffle} package which has ggplot2 offerings like waffle::geom_waffle and which has an underlying stat.  Not suer of overlaps between packages.

**Reply by EvaMaeRey** [reply]
both get mention in this nice tour of extension https://www.cedricscherer.com/slides/RLadiesTunis-2021-favorite-ggplot-extensions.pdf
